unity/UnityEmbedHost.Tests/EmbeddingApiTests.cs

```diff
@@ -55,4 +55,125 @@ public void GCHandleNewAndGetTarget()
         GCHandle.FromIntPtr(handle1).Free();
         GCHandle.FromIntPtr(handle2).Free();
     }
+
+    // Classes and classes
+    [TestCase(typeof(object), typeof(object), true)]
```

Comment:
What runtime are these tests running on? Mono? Our CoreCLR with our GC?

Reply:
From Rider they are going to run on whatever coreclr rider is using.

From the CI scripts, I'm not sure.  Maybe the system installed, maybe our fork.  @UnityAlex do you know?

Reply:
The Unsafe calls making objects into IntPtr's is actually unsafe ;-) on stock CoreCLR as objects may be moved by the GC.

Reply:
> From Rider they are going to run on whatever coreclr rider is using.
> 
> From the CI scripts, I'm not sure. Maybe the system installed, maybe our fork. @UnityAlex do you know?

Yes the way it's setup currently it's just running `dotnet` so whatever is on the PATH env var.

Reply:
> The Unsafe calls making objects into IntPtr's are actually unsafe ;-) on stock CoreCLR as objects may be moved by the GC.

Makes sense.  I didn't think about that.  I guess we'll see how long we get lucky for :smile: